Neos.ContentRepository/Classes/Security/Authorization/Privilege/Node/NodePrivilegeContext.php

```diff
@@ -145,10 +154,18 @@ public function nodeIsOfType($nodeTypes)
     /**
      * Matches if the selected node belongs to one of the given $workspaceNames
      *
-     * Example: isInWorkspace(['live', 'user-admin']) matches if the selected node is in one of the workspaces "user-admin" or "live"
+     * Note: If you want to restrict editing a node if the user is in one of the given workspaces,
+     * use `userIsInTargetWorkspace` instead, as this method only checks the node's workspace.
+     * In general most nodes are in the "live" workspace until they are changed, so this will
+     * only apply if User A edits a node in the "live" workspace and pushes the changes to
+     * workspace "workspace-abcd" and User B tries to edit the node in "workspace-abcd".
+     *
+     * Example: isInWorkspace(['workspace-abcd', 'user-admin', 'live'])
+     * matches if the selected node is in one of the workspaces "workspace-abcd", "user-admin" or "live"
      *
-     * @param array $workspaceNames An array of workspace names, e.g. ["live", "user-admin"]
+     * @param array $workspaceNames An array of workspace names, e.g. ["live", "user-admin", "workspace-abcd"]
      * @return boolean true if the selected node matches the $workspaceNames, otherwise false
+     * @deprecated since 9.0, use userIsInTargetWorkspace instead
```

Comment:
Context: This PR targets the 8.3 branch. But I agree not to deprecated it in a patchlevel release

Reply:
the whole class will cease to exist in Neos 9 so we can literally do anything :D

Reply:
I will remove this and will have a look, what has changed for Neos 9.